20_2 TreeList Column Reordering: rewrite (#795)
DevExpress/devextreme-demos: JSDemos/Demos/TreeList/Reordering/description.md

> The initial order of columns mirrors the order of fields in the first object from the [dataSource](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/#dataSource). You can change that order programmatically and allow users to reorder columns in the TreeList's interface.

Reviewer: The first sentence is not true, see the screenshot. In my opinion, the whole concept is misleading. Are there many cases when users create a TreeList without the **columns** property?

![image](https://user-images.githubusercontent.com/33040934/116368813-63f76480-a811-11eb-8dc3-735bc467d23c.png)

**Properties to reorder columns programmatically:**

> * [columns[]](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/)           

Reviewer: ```suggestion
* **[columns[]](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/)**           
```

> Configures columns. Arrange column names and objects in the desired order.

Reviewer: ```suggestion
Configures columns. Arrange column names or data objects in the desired order.
```

* **columns[]**.[visibleIndex](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/#visibleIndex)          

> Specifies the position of the column regarding other columns. For details, see the [Column Reordering](https://js.devexpress.com/Documentation/Guide/UI_Components/TreeList/Columns/Column_Reordering/) article.

Reviewer: ```suggestion
Specifies the position of the column regarding other columns. For details, see the [Column Reordering](/Documentation/Guide/UI_Components/TreeList/Columns/Column_Reordering/) article.
```

> **Property that enables users to reorder columns:**

Reviewer: I'm not sure that a list with only one item is a good idea. I think that it's better to transform the list to a regular paragraph or move the **allowReordering** description to a separate list item.

> The order of TreeList columns follows the order of fields in data objects. You can use the [columns](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/) array to specify a different order. To reorder a column at runtime, change its [visibleIndex](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/#visibleIndex) property.

Reviewer 2: ```suggestion
TreeList columns have the same order as fields in data objects. You can use the [columns](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/) array to specify a different order. To reorder a column at runtime, change its [visibleIndex](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/#visibleIndex) property.
```

> The order of TreeList columns is the same as the order of fields in data objects. You can use the [columns](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/) array to specify a different order. To reorder a column at runtime, change its [visibleIndex](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/#visibleIndex) property.

Reviewer 3: ```suggestion
The order of TreeList columns follows the order of fields in data objects. You can use the [columns](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/) array to specify a different order. To reorder a column at runtime, change its [visibleIndex](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/#visibleIndex) property.
```

> The structure of [dataSource](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/#dataSource) objects defines the order of TreeList columns. You can change the order programmatically and allow users to reorder columns in the UI.

Reviewer 4: > The structure of [dataSource](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/#dataSource) objects defines the order of TreeList columns.

First the old ("the order of columns"), then the new ("is defined by the structure of data objects").

Also, there's little information to divide the description into subheadings, so I moved information from the following paragraph here.

```suggestion
The order of TreeList columns is the same as the order of fields in data objects. You can use the [columns](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/) array to specify a different order. To reorder a column at runtime, change its [visibleIndex](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/#visibleIndex) property.
```

> 
> ### Reorder columns programmatically
> To reorder columns in code, arrange objects and column names as strings in the desired order inside the [columns[]](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/) array. Additionally, you can use the **columns[]**.[visibleIndex](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/#visibleIndex) property to specify the position of the column regarding other columns. For details, see the [Column Reordering](/Documentation/Guide/UI_Components/TreeList/Columns/Column_Reordering/) article.

Reviewer 4: ```suggestion
```

> To reorder columns in code, arrange objects and column names as strings in the desired order inside the [columns[]](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/) array. Additionally, you can use the **columns[]**.[visibleIndex](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/#visibleIndex) property to specify  the position of the column regarding other columns. For details, see the [Column Reordering](/Documentation/Guide/UI_Components/TreeList/Columns/Column_Reordering/) article.

Reviewer: ```suggestion
To reorder columns in code, arrange objects and column names as strings in the desired order inside the [columns[]](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/) array. Additionally, you can use the **columns[]**.[visibleIndex](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/#visibleIndex) property to specify the position of the column regarding other columns. For details, see the [Column Reordering](/Documentation/Guide/UI_Components/TreeList/Columns/Column_Reordering/) article.
```

> ### Allow users to reorder columns in the UI
> To enable column reorder in the UI, set [allowColumnReordering](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/#allowColumnReordering) to **true**. If some column should never be reordered, set its [allowReordering](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/#allowReordering) property to **false**.

Reviewer 4: I added how users reorder columns and change this bit:

> If some column should never be reordered

to make clear what exactly happens when you disable a column's `allowReordering`.

```suggestion
Users can drag and drop column headers to reorder columns. To enable this feature, set the [allowColumnReordering](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/#allowColumnReordering) property to **true**. If you do not want users to drag a specific column, disable its [allowReordering](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/#allowReordering) property. In this demo, users cannot drag the Full Name column.
```
